tests/test_readme_util.py

```diff
@@ -0,0 +1,337 @@
+
+
+@pytest.mark.parametrize(
+    "readme_md, expected_error",
+    [
+        (README_NO_YAML, EXPECTED_ERROR_README_NO_YAML),
+        (README_EMPTY_YAML, EXPECTED_ERROR_README_EMPTY_YAML),
+        (README_INCORRECT_YAML, EXPECTED_ERROR_README_INCORRECT_YAML),
+        (README_EMPTY, EXPECTED_ERROR_README_EMPTY),
+        (README_NONE_SUBSECTION, EXPECTED_ERROR_README_NONE_SUBSECTION),
+        (README_MISSING_FIRST_LEVEL, EXPECTED_ERROR_README_MISSING_FIRST_LEVEL),
+        (README_MISSING_SUBSECTION, EXPECTED_ERROR_README_MISSING_SUBSECTION),
+        (README_MISSING_TEXT, EXPECTED_ERROR_README_MISSING_TEXT),
+        (README_MULTIPLE_SAME_HEADING_1, EXPECTED_ERROR_README_MULTIPLE_SAME_HEADING_1),
+        (README_WRONG_FIRST_LEVEL, EXPECTED_ERROR_README_WRONG_FIRST_LEVEL),
+        (README_MULTIPLE_WRONG_FIRST_LEVEL, EXPECTED_ERROR_README_MULTIPLE_WRONG_FIRST_LEVEL),
+    ],
+)
+def test_readme_from_string_errors(readme_md, expected_error):
+    with pytest.raises(ValueError, match=expected_error.format(path="root")):
+        ReadMe.from_string(readme_md, example_yaml_structure)
+
+
+def test_readme_from_readme_correct():
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        path = Path(tmp_dir) / "README.md"
+        with open(path, "w+") as readme_file:
+            readme_file.write(README_CORRECT)
+        out = ReadMe.from_readme(path, example_yaml_structure).to_dict()
+        assert out["name"] == path
+        assert out["text"] == ""
+        assert out["is_empty"]
+        assert out["subsections"] == CORRECT_DICT["subsections"]
+
+
+@pytest.mark.parametrize(
+    "readme_md, expected_error",
+    [
+        (README_NO_YAML, EXPECTED_ERROR_README_NO_YAML),
+        (README_EMPTY_YAML, EXPECTED_ERROR_README_EMPTY_YAML),
+        (README_INCORRECT_YAML, EXPECTED_ERROR_README_INCORRECT_YAML),
+        (README_EMPTY, EXPECTED_ERROR_README_EMPTY),
+        (README_NONE_SUBSECTION, EXPECTED_ERROR_README_NONE_SUBSECTION),
+        (README_MISSING_FIRST_LEVEL, EXPECTED_ERROR_README_MISSING_FIRST_LEVEL),
+        (README_MISSING_SUBSECTION, EXPECTED_ERROR_README_MISSING_SUBSECTION),
+        (README_MISSING_TEXT, EXPECTED_ERROR_README_MISSING_TEXT),
+        (README_MULTIPLE_SAME_HEADING_1, EXPECTED_ERROR_README_MULTIPLE_SAME_HEADING_1),
+        (README_WRONG_FIRST_LEVEL, EXPECTED_ERROR_README_WRONG_FIRST_LEVEL),
+        (README_MULTIPLE_WRONG_FIRST_LEVEL, EXPECTED_ERROR_README_MULTIPLE_WRONG_FIRST_LEVEL),
+    ],
+)
+def test_readme_from_readme_error(readme_md, expected_error):
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        path = Path(tmp_dir) / "README.md"
+        with open(path, "w+") as readme_file:
+            readme_file.write(readme_md)
+        expected_error = expected_error.format(path=path).encode("unicode_escape").decode("ascii")
+        with pytest.raises(ValueError, match=expected_error):
+            ReadMe.from_readme(path, example_yaml_structure)
```

Comment:
`match` is supposed to be a regex, however you are passing a path that may be a windows path.
Instead of espacing the backslashes from windows, you can just escape the full string so that it will consider it a a simple litteral.

```suggestion
        expected_error = expected_error.format(path=path)
        with pytest.raises(ValueError, match=re.escape(expected_error)):
            ReadMe.from_readme(path, example_yaml_structure)
```